comms/src/connection_manager/listener.rs

```diff
@@ -127,6 +129,53 @@ where
         }
     }
 
+    async fn spawn_listen_task(&self, socket: TTransport::Output, peer_addr: Multiaddr) {
+        let executor = self.executor.clone();
+        let node_identity = self.node_identity.clone();
+        let peer_manager = self.peer_manager.clone();
+        let mut conn_man_notifier = self.conn_man_notifier.clone();
+        let noise_config = self.noise_config.clone();
+        let our_supported_protocols = self.our_supported_protocols.clone();
+
+        // This will block (asynchronously) if we have reached the maximum simultaneous connections, creating
+        // back-pressure on nodes connecting to this node
+        self.bounded_executor
+            .spawn(async move {
+                match Self::perform_socket_upgrade_procedure(
+                    executor,
+                    node_identity,
+                    peer_manager,
+                    noise_config,
+                    conn_man_notifier.clone(),
+                    socket,
+                    peer_addr,
+                    our_supported_protocols,
+                )
+                .await
+                {
```

Comment:
did cargo fmt do this? makes it tough to read because it no longer looks like the brace is part of the `match` statement

Reply:
suggest `).await {`

Reply:
Yup it returns await back to a new line - I've put the result in a variable and rather match on that